### ISSUE TITLE
Unable to Connect Nuxt App and Sanity Presentation/Visual-Editing

### ISSUE THREAD

**Issue by dhiamonkey** [nuxt-modules/sanity#1084]
### Version
```
Nuxt version; 3.13 and ^3.14
Nuxt/sanity version: 1.12.2
Sanity version: 3.48 and ^3.64
```

## Reproduction

**Link:** 
[My Public Repository of minimal reproduction ](https://github.com/dhiamonkey/nuxt-sanity-presentation-bug) 

#### What is expected?

I can change the title of the placeholder `event` document inside presentation-panel/visual-editing

#### What is actually happening?

Unable to connect frontend app with sanity studio presentation panel.

```
createConnect.js:53 WebSocket connection to 'wss://[project-id].api.sanity.io/v2022-06-30/socket/production?tag=sanity.studio' failed: WebSocket is closed before the connection is established.
```
and
```
Unable to connect to visual editing. Make sure you've setup '@sanity/visual-editing' correctly
```
#### Steps to reproduce
1. Clone said repo
2. Make a new sanity project, get a new sanity project id
3. Create a read token, as a prequisite to use `Presentation` tool
4. Update `projectId` and `token` in `sanity.config.ts` and `app/fe-test-presentation/nuxt.config.ts` with yours.
5. In sanity studio, there is a placeholder `Event` document, please add one.
6. Add `presentationTool` in sanity plugin and uncomment the `visualEditing` field inside `app/fe-test-presentation/`
7. Open presentation panel and the console, you should see those two error log mentioned at the Issue Description

## Additional information
This issue is consistently replicatable in my offic project or a fresh sanity studio project and a fresh new nuxt project like the reproduction repo.

There might be [a similiar open issue](https://github.com/nuxt-modules/sanity/issues/1053), the key difference is that issue happens in vercel but mine is replicatable in **localhost**
## Checklist
* [x]  I have tested with the latest Nuxt version and the issue still occurs
* [x]  I have tested with the latest module version and the issue still occurs
* [x]  I have searched the issue tracker and this issue hasn't been reported yet
* [x]  I have made sure my `projectId` and `token` is correct in both `nuxt.config.ts` and `sanity.config.ts`

**Comment by silvio-e** [comment]
@danielroe After upgrading my sites to the latest Nuxt and latest @nuxtjs/sanity they don't work anymore with visual editing mode on. I have also no luck with setting up new sites with visual on.

I tried several combinations of Nuxt and module versions as I had one site still working with 3.13.2 and @nuxtjs/sanity 1.12.1. But after upgrading this one it's completely broken and giving a 500 and complaining that react is missing.

I could try to help, but I see when looking into the module code that there is quite a steep learning curve. But if you have any idea what could be wrong and point me to the code to work with I'll try my best.

**Comment by danielroe** [comment]
@silvio-e  is your error the same as in the issue here?

**Comment by silvio-e** [comment]
@danielroe I try with the latest Nuxt and module version. Change to the "Presentation" view and it starts loading. After a while, without redirecting in the Sanity address bar it opens the site behind a blurred overlay saying:

"Unable to connect, check the browser console for more information."

In the console I find:

`Uncaught SyntaxError: The requested module 'http://localhost:3000/_nuxt/node_modules/.pnpm/react-compiler-runtime@19.0.0-beta-df7b47d-20241124_react@18.3.1/node_modules/react-compiler-runtime/dist/index.js?v=34a3f6f5' doesn't provide an export named: 'c'` in mutations.ts

and 

`Unable to connect to visual editing. Make sure you've setup '@sanity/visual-editing' correctly` in Preview.tsx

**Comment by silvio-e** [comment]
And there is actually a warning too:

`Failed to resolve dependency: @nuxtjs/sanity > @sanity/visual-editing > @sanity/visual-editing > react-is, present in 'optimizeDeps.include'`

**Comment by dhiamonkey** [comment]
Same with @silvio-e, I found that warning too, but installing `react-is` will not solve the issue.

I also encounter some react-compiler issue.

**Comment by rylanharper** [comment]
Hey all, just checking to see if this still an issue with the most recent release?

**Comment by icelandcruiser** [comment]
I get the following error: 

`ERROR  [nuxt] [request error] [unhandled] [500] Cannot find module './cjs/react.development.js'`
`Require stack: /Users/.../react`

If I manually add react and react-dom to my dependencies, I also encounter the exact same error, as described by @silvio-e:

`Uncaught SyntaxError: The requested module '/_nuxt/@fs/Users/.../node_modules/.pnpm/react-compiler-runtime@19.0.0-beta-df7b47d-20241124_react@18.3.1/node_modules/react-compiler-runtime/dist/index.js?v=ea4f85c3' does not provide an export named 'c' (at mutations.js?v=ea4f85c3:1:8)`

**Comment by finnbuster** [comment]
Did this end up being the simple fix you thought it was by any chance?

> I think you're right - this is coming from the sanity package dependency changes.
> 
> should be a quick fix in the module (will try this afternoon) or you can pin the sanity visual editing package in the meantime in case that resolves things

**Comment by danielroe** [comment]
yes, this was a simple fix, but:

1. I'll need to think to see how we can avoid being broken by dependency changes upstream in sanity that pull in non-ESM code into the browser

2. additionally I would like somehow to add an e2e test that covers the flow of enabling visual editing so we can detect this earlier when upgrading dependencies.

**Comment by danielroe** [comment]
this is now fixed in v1.13.1 (as a hotfix)

In future if you experience an issue like this before I can address it in the module, here's how to work around it temporarily:

* get error message about a missing optimise dep
* run `pnpm why <the dependency>` - it should give you a chain of dependencies
* add this chain to your `vite.optimizeDeps.include` array, e.g. `@nuxtjs/sanity > @sanity/visual-editing > @sanity/mutate > lodash/groupBy.js`

if you do this, please raise an issue and let me know.
